### ISSUE TITLE
[css-cascade-5] What happens to name-defining @-rules nested inside @layer?

### ISSUE THREAD

**Issue by andruud** [w3c/csswg-drafts#6404]
For example, you might expect this to define a _fancy_ animation, ignoring the plain one:

```CSS
@layer plain fancy;

@layer fancy {
  @keyframes anim { /* ... fancy animation ... */ }
}

@layer plain {
  @keyframes anim { /* ... plain animation ... */ }
}
```
It might be weird if it defines a _plain_ animation? So probably another reasonable behavior is to make name-defining things 
(e.g. `@keyframes`, `@property`, `@scroll-timeline`, etc) _parse errors_ if they appear inside `@layer`.

cc @mirisuzanne @xiaochengh

**Comment by mirisuzanne** [comment]
What's the full list of @-rules we need to consider here? `@font-face` as well? Are there more? Is it best if we come to the same conclusion for all of them? 

Do we know if people currently use source-order to override name-defining rules? I wonder how much that happens. I think layers can be used in many cases to make source-order less important, so if there's a strong existing use-case for re-defining/overriding these names already, then I'd lean towards applying the layer-order cascade to these rules.

**Comment by SebastianZ** [comment]
A full list of at-rules can be found at https://drafts.csswg.org/indexes/#at-rules. As far as I can see, the name-defining ones are these:

`@color-profile`
`@container`
`@counter-style`
`@custom-media`
`@custom-selector`
`@font-face` (through its `font-family` descriptor)
`@font-palette-values`
`@keyframes`
`@layer` (of course)
`@property`
`@scroll-timeline`

Sebastian

**Comment by mirisuzanne** [comment]
Thanks, @SebastianZ 

I think we can ignore a couple of those here:

- `@container` queries can reference a named container, but doesn't actually define that name. The name can be used as often as desired, without impacting previous usage - so I don't think that one's an issue.
- `@layer` already has a defined behavior for nesting.

For the rest, the main use-case I can imagine is dealing with third-party code: eg overriding the built-in font or animation or counters (etc) from a component library. Though [I don't think that's terribly common right now](https://twitter.com/TerribleMia/status/1407437642715648000), it is one of the primary use-cases for `@layer`, so I do _like_ the idea of having support for it.

**Comment by mirisuzanne** [comment]
Closed by a31095710
